idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionCompletionContributor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.codeInsight.completion.CompletionContributor;
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionProvider;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.CompletionType;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.application.ReadAction;
+import com.intellij.patterns.PlatformPatterns;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ProcessingContext;
+import com.palantir.gradle.versions.intellij.psi.VersionPropsDependencyVersion;
+import com.palantir.gradle.versions.intellij.psi.VersionPropsProperty;
+import com.palantir.gradle.versions.intellij.psi.VersionPropsTypes;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class VersionCompletionContributor extends CompletionContributor {
+
+    private static final Logger log = LoggerFactory.getLogger(VersionCompletionContributor.class);
+
+    VersionCompletionContributor() {
+        extend(
+                CompletionType.BASIC,
+                PlatformPatterns.psiElement(VersionPropsTypes.VERSION),
+                new CompletionProvider<>() {
+                    @Override
+                    public void addCompletions(
+                            @NotNull CompletionParameters parameters,
+                            @NotNull ProcessingContext context,
+                            @NotNull CompletionResultSet resultSet) {
+
+                        VersionPropsDependencyVersion versionElement =
+                                ReadAction.compute(() -> (VersionPropsDependencyVersion)
+                                        parameters.getPosition().getParent());
+
+                        VersionPropsProperty property = ReadAction.compute(() -> findParentProperty(versionElement));
+
+                        DependencyGroup group = DependencyGroup.fromString(
+                                property.getDependencyGroup().getText());
+                        DependencyPackage dependencyPackage = DependencyPackage.of(
+                                property.getDependencyName().getText());
```

Comment:
We should be consistent with the naming - I'd call `DependencyPackage` `DependencyName` instead. To me package means group and name together.

Reply:
good point, not sure why I switched half way through! have refactored